lib/spack/env/cc

```diff
@@ -339,7 +339,11 @@ case "$mode" in
         args=("${args[@]}" ${SPACK_LDLIBS[@]}) ;;
 esac
 
-full_command=("$command" "${args[@]}")
+if [[ ${lang_flags} != "F" && ${SPACK_CCACHE_BINARY} ]]; then
```

Comment:
Add a comment explaining what `${lang_flags}!="F"` means.  I think you're filtering out the Fortran compiler.